api/controllers/comment.js

```diff
@@ -138,116 +205,131 @@ exports.protectedPut = function (args, res, next) {
 exports.protectedUnPublish = function (args, res, next) {
   var objId = args.swagger.params.CommentId.value;
   defaultLog.info("UnPublish Comment:", objId);
 
   var Comment = require('mongoose').model('Comment');
-  Comment.findOne({_id: objId}, function (err, o) {
+  Comment.findOne({ _id: objId }, function (err, o) {
     if (o) {
       defaultLog.info("o:", o);
 
       // Remove public to the tag of this obj.
       Actions.unPublish(o)
-      .then(function (unpublished) {
-        // UnPublished successfully
-        return Actions.sendResponse(res, 200, unpublished);
-      }, function (err) {
-        // Error
-        return Actions.sendResponse(res, err.code, err);
-      });
+        .then(function (unpublished) {
+          // UnPublished successfully
+          return Actions.sendResponse(res, 200, unpublished);
+        }, function (err) {
+          // Error
+          return Actions.sendResponse(res, err.code, err);
+        });
     } else {
       defaultLog.info("Couldn't find that object!");
       return Actions.sendResponse(res, 404, {});
     }
   });
 };
-var getComments = function (role, query, fields) {
+
+var getComments = function (role, query, fields, sort, skip, limit) {
   return new Promise(function (resolve, reject) {
     var Comment = mongoose.model('Comment');
     var projection = {};
 
     // Fields we always return
-    var defaultFields = ['_id',
-                        'tags'];
+    var defaultFields = [
+      '_id',
+      'tags'
+    ];
     _.each(defaultFields, function (f) {
-        projection[f] = 1;
+      projection[f] = 1;
     });
 
     // Add requested fields - sanitize first by including only those that we can/want to return
     var sanitizedFields = _.remove(fields, function (f) {
-      return (_.indexOf(['name',
-                        'commentNumber',
-                        'comment',
-                        'internal',
-                        'dateAdded',
-                        'commentAuthor',
-                        'review',
-                        '_addedBy',
-                        '_commentPeriod',
-                        'review',
-                        'commentStatus',
-                        'isDeleted'], f) !== -1);
+      return (_.indexOf([
+        'name',
+        'commentNumber',
+        'comment',
+        'internal',
+        'dateAdded',
+        'commentAuthor',
+        'review',
+        '_addedBy',
+        '_commentPeriod',
+        'review',
+        'commentStatus',
+        'isDeleted'
+      ], f) !== -1);
     });
     _.each(sanitizedFields, function (f) {
       projection[f] = 1;
     });
 
-    Comment.aggregate([
-      {
-        "$match": query
-      },
-      {
-        "$project": projection
-      },
+    if (!_.isEmpty(sort)) {
+      // extract contactName so we can sort on it
+      // change to lower-case for case-insensitive sorting
+      projection.contactName = { $toLower: "$commentAuthor.contactName" };
```

Comment:
This will cause the system to leak the contact name even if they've requested anonymous and the public tag has been removed on the commentAuthor object.